tests/test_ir.py

```diff
@@ -232,16 +233,37 @@ def test_is_structurally_equivalent_incompatible_ir_nodes():
     ctx.register_dialect(Cf)
 
     parser = Parser(ctx, program_func)
-    program: ModuleOp = parser.parse_op()
+    program: Operation = parser.parse_op()
```

Comment:
Why was this part changed?

Reply:
This fixes a typing error: `parser.parse_op` returns an `Operation`. I missed this in the previous PR and fixed it here